File: vapor-app/src/main/java/com/tevinjeffrey/vapor/ui/files/fragments/FilesFragmentView.java
package com.tevinjeffrey.vapor.ui.files.fragments;

import com.tevinjeffrey.vapor.okcloudapp.DataManager;
import com.tevinjeffrey.vapor.okcloudapp.model.CloudAppItem;
import com.tevinjeffrey.vapor.ui.base.View;

import java.util.List;

public interface FilesFragmentView extends View {
    void showLoading(boolean pullToRefresh);

    void setData(List<CloudAppItem> data);

    void appendData(List<CloudAppItem> data);

    void showError(Throwable e);

    void showLayout(LayoutType type);

    void initRecyclerView();

    void initSwipeLayout();

    DataManager.DataCursor getCursor();
}

File: vapor-app/src/main/java/com/tevinjeffrey/vapor/ui/files/fragments/FilesFragmentView.kt
package com.tevinjeffrey.vapor.ui.files.fragments

import com.tevinjeffrey.vapor.okcloudapp.DataManager
import com.tevinjeffrey.vapor.okcloudapp.model.CloudAppItem
import com.tevinjeffrey.vapor.ui.base.View

interface FilesFragmentView : View {
    fun showLoading(pullToRefresh: Boolean)

    fun setData(data: List<CloudAppItem>)

    fun appendData(data: List<CloudAppItem>)

    val isVisibleInPager: Boolean

    fun showError(e: Throwable)

    fun showLayout(type: View.LayoutType)

    fun initRecyclerView()

    fun initSwipeLayout()

    val cursor: DataManager.DataCursor
}
